src/test/java/net/minecraftforge/debug/entity/living/ExpandedElytraFlightTest.java

```diff
@@ -0,0 +1,221 @@
+    private static final String FLIGHT_SPEED_EFFECT_MODIFIER_UUID_STRING = "41fddd0b-1ff1-4f24-87a9-b3f23e420187";
+    private static final EquipmentSlotType[] ARMOR_SLOTS = new EquipmentSlotType[]{EquipmentSlotType.HEAD, EquipmentSlotType.CHEST, EquipmentSlotType.LEGS, EquipmentSlotType.FEET};
+
+    // Effects
+    private static final RegistryObject<Effect> FLIGHT_EFFECT = EFFECTS.register("flight",
+            () -> new FallFlightEffect(EffectType.BENEFICIAL, 0x0000FF)
+                    .addAttributeModifier(ForgeMod.FALL_FLIGHT.get(), FLIGHT_EFFECT_MODIFIER_UUID_STRING, FLIGHT_EFFECT_MODIFIER_VALUE_ADDITION, AttributeModifier.Operation.ADDITION)
+    );
+
+    private static final RegistryObject<Effect> NO_FLIGHT_EFFECT = EFFECTS.register("no_flight",
+            () -> new FallFlightEffect(EffectType.HARMFUL, 0xFF0000)
+                    .addAttributeModifier(ForgeMod.FALL_FLIGHT.get(), NO_FLIGHT_EFFECT_MODIFIER_UUID_STRING, -FLIGHT_EFFECT_MODIFIER_VALUE_ADDITION, AttributeModifier.Operation.ADDITION)
+    );
+
+    // Potions
+    private static final RegistryObject<Potion> FLIGHT_POTION = POTIONS.register("flight",
+            () -> new Potion(new EffectInstance(FLIGHT_EFFECT.get(), FLIGHT_POTION_EFFECT_DURATION))
+    );
+
+    private static final RegistryObject<Potion> NO_FLIGHT_POTION = POTIONS.register("no_flight",
+            () -> new Potion(new EffectInstance(NO_FLIGHT_EFFECT.get(), FLIGHT_POTION_EFFECT_DURATION))
+    );
+
+    // Enchantments
+    private static final RegistryObject<Enchantment> FLIGHT_ENCHANTMENT = ENCHANTMENTS.register("flight",
+            () -> new FlightEnchantment(Enchantment.Rarity.VERY_RARE, EnchantmentType.ARMOR, ARMOR_SLOTS)
+    );
+
+    private static final RegistryObject<Enchantment> FLIGHT_SPEED_ENCHANTMENT = ENCHANTMENTS.register("flight_speed",
+            () -> new FlightEnchantment(Enchantment.Rarity.VERY_RARE, EnchantmentType.ARMOR, ARMOR_SLOTS)
+    );
+
+    // Items
+    private static final RegistryObject<Item> FLIGHT_BOOTS = ITEMS.register("flight_boots",
+            () -> new FlightArmorItem(ArmorMaterial.IRON, EquipmentSlotType.FEET, true, FLIGHT_SPEED_ARMOR_MODIFIER_VALUE_ADDITION, (new Item.Properties()).tab(ItemGroup.TAB_COMBAT)));
+
+    private static final RegistryObject<Item> NO_FLIGHT_LEGGINGS = ITEMS.register("no_flight_leggings",
+            () -> new FlightArmorItem(ArmorMaterial.IRON, EquipmentSlotType.LEGS, false, 0, (new Item.Properties()).tab(ItemGroup.TAB_COMBAT)));
+
+    public ExpandedElytraFlightTest()
+    {
+        if(ENABLE){
+            IEventBus modEventBus = FMLJavaModLoadingContext.get().getModEventBus();
+            EFFECTS.register(modEventBus);
+            POTIONS.register(modEventBus);
+            ENCHANTMENTS.register(modEventBus);
+            ITEMS.register(modEventBus);
+            MinecraftForge.EVENT_BUS.register(this);
+        }
+    }
+
+    @SubscribeEvent
+    public void onLivingUpdate(LivingEvent.LivingUpdateEvent event)
```

Comment:
You could use the item attribute event here for the record, cleaner than manually changing the attributes. Also means it will show up in the tooltip, to be a good test case for how these boolean attributes display.

Reply:
The modifiers added here during this event are for the enchantments, using the same logic as how vanilla adds the Soul Speed enchantment's modifier to the player.